src/test/java/store/mybookstore/service/shoppingcart/impl/ShoppingCartServiceImplTest.java

```diff
@@ -0,0 +1,143 @@
+package store.mybookstore.service.shoppingcart.impl;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.when;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+import store.mybookstore.dto.cartitem.CartItemRequestDto;
+import store.mybookstore.dto.cartitem.CartItemResponseDto;
+import store.mybookstore.dto.shoppingcart.ShoppingCartResponseDto;
+import store.mybookstore.model.Book;
+import store.mybookstore.model.CartItem;
+import store.mybookstore.model.Category;
+import store.mybookstore.model.ShoppingCart;
+import store.mybookstore.model.User;
+import store.mybookstore.repository.ShoppingCartRepository;
+import store.mybookstore.service.cartitem.CartItemService;
+import store.mybookstore.service.user.UserService;
+
+@ExtendWith(MockitoExtension.class)
+class ShoppingCartServiceImplTest {
+    private static Long id;
+    private static User user;
+    private static ShoppingCart shoppingCart;
+    private static ShoppingCartResponseDto shoppingCartResponseDto;
+    private static CartItem cartItem;
+    private static CartItemResponseDto cartItemResponseDto;
+    private static CartItemRequestDto cartItemRequestDto;
+    private static Book book;
+    private static Category category;
+    @Mock
+    private UserService userService;
+    @Mock
+    private ShoppingCartRepository shoppingCartRepository;
+    @Mock
+    private CartItemService cartItemService;
+    @Mock
+    private ShoppingCartManager shoppingCartManager;
+    @InjectMocks
+    private ShoppingCartServiceImpl shoppingCartService;
+
+    @BeforeAll
+    static void beforeAll() {
+        user = new User();
+        user.setRoles(Set.of());
+        user.setId(id);
+        user.setFirstName("Bob");
+        user.setLastName("Bober");
+        user.setEmail("bob@example.com");
+        user.setPassword("12345678");
+        user.setShippingAddress("Vilna 7");
+
+        category = new Category();
+        category.setId(id);
+        category.setName("test_category");
+
+        book = new Book();
+        book.setId(id);
+        book.setAuthor("Author A");
+        book.setTitle("Title A");
+        book.setDescription("Description A");
+        book.setPrice(BigDecimal.valueOf(10));
+        book.setIsbn("10001000");
+        book.setCoverImage("https://example.com/default-cover-image.jpg");
+        book.setCategories(Set.of(category));

```

Comment:
Please extract logic of creating objects somewhere (private methods of this class or separate TestData class if you plan to create similar objects in other tests)
At the moment whole first half of test class is creation of test required objects
```suggestion
        user = getDefaultUser();
        category = getTestCategory();
        book = getBookUnderCategory(category);

```